pydantic/_internal/_generate_schema.py

```diff
@@ -2289,3 +2294,22 @@ def get(self) -> str | None:
             return self._stack[-1]
         else:
             return None
+
+
+class _ModelTypeStack:
+    __slots__ = ('_stack',)
+
+    def __init__(self) -> None:
+        self._stack: list[str] = []
+
+    @contextmanager
+    def push(self, type_obj: str) -> Iterator[None]:
+        self._stack.append(type_obj)
+        yield
+        self._stack.pop()
+
+    def get(self) -> str | None:
```

Comment:
Could we call this `tail` to be consistent with the other stack implementations used in the generate schema logic?

Reply:
I see you've modeled this on `_FieldNameStack` from above, so you can disregard this - `get` is fine!